### DISCUSSION TITLE
O Desenvolvedor empreendedor

### DISCUSSION THREAD

**Discussion by Cleysonlb** [frontendbr/forum#111] [Discussão]
Olá meus amigos tudo bem?! Venho trazer esse papo sobre o desenvolvedor empreendedor, não é regra, mas na grande maioria os desenvolvedores tem um "Q" de empreendedor, seja ele sendo freelancer, criando produtos ou empreendendo dentro da empresa em que trabalha, bom, criei essa Issue para dividir experiências, ideias, etc... :)

**Comment by felipefialho** [comment]
Fala @Cleysonlb, beleza?

Conta um pouco como seria empreender dentro da empresa em que trabalha?

Fiquei curioso.

**Comment by raphaklaus** [comment]
Ótimo tópico!!

Bom, vou deixar minha visão e experiência nesse mundo muito **loko**! Estou longe de ser guru, porém com algumas pequenas vitórias e alguns fracassos, sempre dá pra compartilhar coisas interessantes!

## Intra empreendedorismo (empreender dentro da empresa)

Considero uma das melhores formas de se **aprender** a empreender. Porque você estará num ambiente "controlado" e com certas "seguranças", salário caindo final do mês, pessoal te orientando e dando feedback. Porém não adianta nada tentar empreender dentro de uma empresa que tem líderes de mente fraca e egoístas de liderança... Você vai acabar criando **inimizades**. 

Se der certo, será um grande laboratório e você terá aprendido demais. 

## Empreendedorismo de fato

É algo bem desafiador quando você é um programador, porque geralmente programadores são mais fechados (não é regra) no código, deixando de lado a visão macro do produto, estratégias de marketing, etc. Vou deixar uma frase que escutei de um investidor em um evento que fui:

> Daqui a 20 anos todos os CEO's serão programadores.

Tá, talvez ele tenha colocado muitos anos aí, mas é a pura verdade. Nós programadores/gerentes de dev/CTO's temos uma responsabilidade grande. Eu já vi muito CEO ficar na mão de CTO por questões técnicas. Pessoas non-techs não entendem certas impossibilidades e algumas vezes fazem a empresa entrar em um colapso interno. 

Acredito que programadores tem um dom especial para aprender as coisas de forma rápida e eficiente. Isso não seria diferente com matérias como: marketing, administração, negócios, é só ter o mindset correto.

## Minha experiência

Ano passado eu tive a oportunidade de ir em vários eventos aqui no RJ e em SP. Conheci muita gente. O que tem de pessoas procurando um programador-escravo (sócio) para sua ideia de um bilhão de dólares, não tá no gibi... 

Na boa? **Não caiam nessa...** Ideia vocês também tem! O cara entrar com a ideia e você entrar com o trabalho é o tipo de relação mais injusta. Geralmente a pessoa que entra com a ideia não tem muito conhecimento necessário para tocar um negócio. Às vezes o cara só é criativo, e ponto.

Hoje em dia dificilmente eu teria sócios. Só se o cara fosse espetacular. Eu preferiria tirar dinheiro do meu bolso e pagar algum funcionário/freelancer pra tocar a parte de programação e eu ficar com o produto/negócios.

Outra coisa que eu não faria de novo é: **bootstrapping.** É difícil de dar certo, as pessoas acabam deixando pra lá, fazem sem muito foco. Quer ver algo andar com mais voracidade? Injete dinheiro no negócio. As pessoas tendem a dar seu melhor para terem a impressão de que o dinheiro valeu à pena.

## 

Desculpe o texto longo demais, mas esse tipo de assunto sempre dá o que falar. :P

**Comment by Cleysonlb** [comment]
Ótimo texto @raphaklaus tive algumas experiências falhas com esses caras com ideias de "um milhão", até tive um sócio que sacaneou bonito comigo.... Mas, Tudo é experiência, certo?!
Respondendo ao @LFeh, não tive oportunidade de empreender em empresa pois a algum tempo trabalho como freelancer (oq pode ser considerado como empreender?!) mas tenho um amigo que criou um produto dentro e usando recursos da empresa onde ele se tornou sócio do produto ao seu término, achei bem interessante, mas é como o @raphaklaus falou, se for empreender em empresas de mente fechada corre o risco de só conseguir inimizades.

**Comment by flpms** [comment]
Eu sou programador empreendedor. E vou contar algumas experiencias que já tive. Já passei por algumas consultorias e por empresas com seus produtos próprios. 

Uma das empresas que empreendi foi dentro do Apontador, e foi uma experiencia incrivel. A fundação do Apontador é de um programador logo o ambiente criado foi mto legal e incentivava muito essa postura na época.

A oportunidade de sugerir ideias e leva-las até o final executando é espetacular. Mas precisa de abertura para isso aqui vai um outro ponto. Logo depois disso entrei em consultoria e fui trabalhar para um cliente do setor financeiro de grande porte. Lá foi com a mesma empolgação, mas foram tantas travas e amarras que você acaba brigando tds os dias para ver as coisas acontecerem de forma correta (Não nem para acontecer de forma além, é de forma correta mesmo).

Após um tempo resolvi empreender em minha [empresa](https://local1202) poder escolher tecnologias e ver o produto nascendo é incrível. Infelizmente o projeto ainda está no _bootstraping_, não recomendo para ninguém. Tem dias que dá uma agonia ver o projeto precisando de tempo ou dinheiro e você estar sem os dois. Fora que geralmente trabalho nele 20hrs por semana contra 40hrs para as outras empresas q trabalho.

Quanto as outras empresas procuro e continuo empreendendo, dentro delas. A mentalidade do empreendedor conta mto para isso e a possibilidade de entrega como citado pelo @raphaklaus.

**Comment by symonlopes** [comment]
Tópico antigo mas me senti na obrigação de dar algumas palavras sobre o tema :)

Já empreendi sozinho, e dentro de empresa.

**Dentro da empresa** 

Muitas vezes a natureza do negócio da empresa que você trabalha simplesmente não permite que seu empreendedorismo seja acolhido, já imaginou se todo mundo da equipe quisesse colocar suas ideias "maravilhosas" dentro da estrutura da empresa? Para isso acontecer, a empresa teria que criar um programa com verba e tempo dedicado a isso, com ganhos reais para ambos os lados. Foi isso que o Google fez e conseguiu alguns bons produtos, como o GMAIL por exemplo; mas também teve grandes  fracassos como o Knol e Google Wave  - um dos maiores fiascos que vi na minha vida.

Fora que empreender dentro de empresa é como andar de bicicleta usando rodinhas, você até anda, mas na hora que tirá-las, você pode ter andado 3000 km e ainda assim não vai estar preparado.

**Fora da empresa**

Programador sente-se confortável na frente do computador, quando vai testar um framework novo, usar um recurso novo do HTML5, e as vezes isso é de fato inútil do ponto de vista do negócio como um todo; além disso, o que para mim é o pior dos problema, é o fato de você ficar sempre "melhorando" o sistema. Todos aqueles refactories e atualizações de bibliotecas que seu chefe cortava e você ficava puto, você terá a liberdade de fazê-los, da forma que te der na telha, porém, com o tempo você vai ver que não é bem assim, que antes do seu primor técnico, a sobrevivência do negócio é muito mais relevante, e se for teimoso e ficar brincando de atualizar framework enquanto as necessidades reais do negócio vão passando, é capaz de não durar 2 anos.

Fora as dificuldades inerentes aos costumes da nossa profissão, você ainda tem que enfrentar todas as outras dificuldades genéricas que todo empreendedor passa: governo como um parasita te sugando, relacionamentos VS empreendedorismo, falta de dinheiro, falta de tempo, stress, e, pessoas.

**Comment by rafaelnct** [comment]
Olá, não sei se alguem vai ler isso, mas aonde posso achar programadores que QUEIRAM de fato empreender, criar algo, enfim que tenham real vontade de programar e queiram criar uma amizade e confiança hehe, existe algum ambiente online pra isso?

Acabei caindo aqui nessa conversa justamente tentando procurar isso, eu precisava de alguem que programasse junto comigo uma ideia que alias já tenho BOA PARTE desenvolvida de certa forma, porem precisava de um auxílio, uma ajuda, aonde poderia buscar essa ajuda, tanto AJUDA pondo a mão na massa (um outro programador) quanto uma ajuda mais talvez teórica ou verbal, enfim... abçs e obrigado pela atenção!!!

**Comment by kazzkiq** [comment]
@rafaelnct O [CoFoundersLabs](https://cofounderslab.com/) é um bom serviço pra isso. Tem bastante brasileiros lá e já consegui achar devs interessados em participar de projetos até na minha cidade!

Fora isso você pode tentar entrar nos grupos de frontend do Facebook ou até mesmo criar uma Issue aqui no fórum com um briefing do projeto e que tipo de ajuda necessita de um dev front-end, acredito que uma Issue dessas não vá contra as regras da comunidade (@felipefialho me corrija se eu tiver falado bobagem :-P).
